README.md

```diff
@@ -2217,4 +2217,219 @@ A Japanese word meaning "form", relating to karate. Used commonly in programming
 
 An excellent example of a Test Driven Development Kata:
 
-[The Bowling Game](http://butunclebob.com/ArticleS.UncleBob.TheBowlingGameKata)
\ No newline at end of file
+[The Bowling Game](http://butunclebob.com/ArticleS.UncleBob.TheBowlingGameKata)
+
+## Practice Interview Questions
+
+### Markdown
+
+- Easy: How do you write bullet points in Markdown?
+- Medium: How do you write check boxes in Markdown? If you write this correctly, and it still doesn't work, what do you think is the issue? (There are different flavours of Markdown and the GitHub version has some differences).
+- Hard: How do you write a table in Markdown?
+
+### Git
+
+- Easy: Assume you've changed a file called `MyClass.java` in a repo your computer. Walk me through the steps of how to get that change to show up on GitHub.
+- Medium: How do you discard all the changed files on your current branch locally. Including completely new files?
+- Hard: Suppose you've accidentally pushed to master. How can you undo that change, but keep all the work on a separate branch. Walk me through all the steps.
+
+### Bash
+
+- Easy: What does the `cat` command do?
+- Medium: What is the $PATH variable?
+- Hard: How can you make it so that whenever you execute 'check', you run both checkstyle and all the tests of the gradle project you're currently in.
+
+### OOP
+
+- Easy: What is inheritance?
+- Medium: What are the downsides of using inheritance?
+- Hard: How can you replace a switch statement with polymorphism?
+
+### TDD:
+
+- Easy: What are the 'Red/Green/Refactor' steps?
+- Medium: What are the benefits of TDD?
+- Hard: What are the downsides of TDD?
+
+### Design
+
+- Easy: What is the difference between design and documentation? How are they similar?
+- Medium: Describe a design process you think will work for a software project.
+- Hard: What concerns do you have to keep in mind when working on a team that is working on several projects at the same time.
+
+## Internet Browsers
+
+There are 3 languages that form what we see when we look at internet browsers.
+
+HTML (Hypertext Markup Language) forms the structure of web pages.
+CSS (Cascading Style Sheets) forms the style.
+JavaScript provides the dynamic and interactive parts.
+
+### JavaScript
+
+JavaScript started as a browser language, but now it is used for a variety of things including: Slack, Facebook Messenger, Web Servers and Applications.
+
+The JavaScript execution engine is Node. No HTML is required, but CSS is.
+
+As JavaScript does not have types, it does not have type safety. Microsoft addressed the issues of this by creating TypeScript, which includes all the capabilities of JavaScript and additionally includes types.
```

Comment:
Javascript does have types (in fact you list the important ones below ;)). Only very few, really odd programming languages (e.g. bash) are untyped.

```suggestion
As JavaScript does not have a compile-time type system, and as such it does not have type safety. Microsoft addressed the issues of this by creating TypeScript, which includes all the capabilities of JavaScript and additionally includes types.
```